pkg/issuer/acme/dns/pdns/pdns.go

```diff
@@ -0,0 +1,356 @@
+// Package pdns implements a DNS provider for solving the DNS-01
+// challenge using PowerDNS.
+package pdns
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+    "bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/dns/v1"
+
+	"github.com/jetstack/cert-manager/pkg/issuer/acme/dns/util"
+)
+
+// DNSProvider is an implementation of the DNSProvider interface.
+type DNSProvider struct {
+    apiKey     string
+	host       *url.URL
+	apiVersion int
+	client     *http.Client
+}
+
+// NewDNSProvider returns a DNSProvider instance configured for pdns.
+// Credentials must be passed in the environment variable:
+// PDNS_API_URL and PDNS_API_KEY.
+func NewDNSProvider() (*DNSProvider, error) {
+    values, err := os.Get("PDNS_API_KEY", "PDNS_API_URL")
+	if err != nil {
+		return nil, fmt.Errorf("PDNS: %v", err)
+	}
+
+	hostURL, err := url.Parse(values["PDNS_API_URL"])
+	if err != nil {
+		return nil, fmt.Errorf("PDNS: %v", err)
+	}
+
+	return NewDNSProviderCredentials(hostURL, values["PDNS_API_KEY"])
+}
+
+// NewDNSProviderCredentials uses the supplied credentials to return a
+// DNSProvider instance configured for pdns.
+func NewDNSProviderCredentials(host *url.URL, key string) (*DNSProvider, error) {
+    if key == "" {
+		return nil, fmt.Errorf("PDNS API key missing")
+	}
+
+	if host == nil || host.Host == "" {
+		return nil, fmt.Errorf("PDNS API URL missing")
+	}
+
+	d := &DNSProvider{
+		host:   host,
+		apiKey: key,
+		client: &http.Client{Timeout: 30 * time.Second},
+	}
+
+	apiVersion, err := d.getAPIVersion()
+	if err != nil {
+		log.Warnf("PDNS: failed to get API version %v", err)
```

Comment:
replace these calls to `log` with `glog` (you'll also need to import `github.com/golang/glog`)